### PR TITLE
File Extraction

### PR DESCRIPTION
**Pull request by dbcfd**
Add support for enabling suricata file extraction.

### CODE REVIEW

**Comment by JPercivall** [comment]
The ouputs listed here[1], we don't need to add a new one for FileInfo?

[1] https://github.com/protectwise/suricata-ipc/pull/33/files#diff-b0ba3e54b6f5ef1fa6aed8075e352a68fda66067fbafa8421042a2347801338aR163

**Comment by dbcfd** [comment]
> The ouputs listed here[1], we don't need to add a new one for FileInfo?
> 
> [1] https://github.com/protectwise/suricata-ipc/pull/33/files#diff-b0ba3e54b6f5ef1fa6aed8075e352a68fda66067fbafa8421042a2347801338aR163

Added here https://github.com/protectwise/suricata-ipc/pull/33/files#diff-29edda423cb602876706ea17ca58531b45401f5141999632926eeb8dde628edeR90

File: src/eve/message/file.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum FileState {
+    #[serde(rename = "OPEN")]
+    Open,
+    #[serde(rename = "CLOSED")]
+    Closed,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct FileInfo {
```
Comment (JPercivall) [inline src/eve/message/file.rs:12]: Doesn't this need to account for the potential hash fields?

Also, is there anything need for the "force-magic" config?

Comment (JPercivall) [reply src/eve/message/file.rs:12]: Actually, I guess these would also need to be exposed here[1]?

[1] https://github.com/protectwise/suricata-ipc/pull/33/files#diff-29edda423cb602876706ea17ca58531b45401f5141999632926eeb8dde628edeR104

Comment (dbcfd) [reply src/eve/message/file.rs:12]: Wasn't going to add file hash support to this PR. Could do another PR if that is desired. 

Magic is currently defaulted to off. Same thing, could do a follow on PR to log magic.

Comment (JPercivall) [reply src/eve/message/file.rs:12]: Rgr, we'll definitely need at least one hash of the file but looking at the docs, it seems the file-id used to write to disk is the SHA256. Is that the "sid" in the fileInfo?

Comment (dbcfd) [reply src/eve/message/file.rs:14]: To actually store them on disk, you need a rule https://github.com/protectwise/suricata-ipc/pull/33/commits/d1516c13d5d2660d8b22931999f1dea2f179318a#diff-10175120aeda1f9b3962be0bcbf25906e6e88636bd37e54fdb04cfaee961e906R23

https://suricata.readthedocs.io/en/suricata-6.0.0/file-extraction/file-extraction.html#rules

I'll add support for hash.

Comment (dbcfd) [reply src/eve/message/file.rs:14]: https://github.com/protectwise/suricata-ipc/pull/33/commits/6dda6f1924db39acf3825546deade089facf5ad4

**Review by JPercivall — APPROVED** [review]
+1 Changes look gtg. I have some issues running locally b/c of events getting dropped due to slow box but that's a known issue and Github works.

**Pull request merged**